### ISSUE TITLE
visitRepeatStmt method needs revision

### ISSUE THREAD

**Issue by VinArt** [OpenRoberta/openroberta-lab#298]
Repeat statement fails to validate its input for the depth > 1.

So this works:
![image](https://user-images.githubusercontent.com/3738015/68580327-6bd51480-0476-11ea-8294-e2254dea3f39.png)

but this already does not:
![image](https://user-images.githubusercontent.com/3738015/68580351-7a233080-0476-11ea-927c-76a961d65b0b.png)

Because we're plugging in a Unary expression and this is how it's dealt with:

```
if ( repeatStmt.getExpr() instanceof EmptyExpr ) {
    repeatStmt.addInfo(NepoInfo.error("ERROR_MISSING_PARAMETER"));
    this.errorCount++;
} else if ( repeatStmt.getExpr() instanceof Unary ) {
    if ( ((Unary<Void>) repeatStmt.getExpr()).getExpr() instanceof EmptyExpr ) {
        repeatStmt.addInfo(NepoInfo.error("ERROR_MISSING_PARAMETER"));
        this.errorCount++;
    }
}
```

Which should be something like this:

```
if ( repeatStmt.getExpr() instanceof EmptyExpr ) {
    repeatStmt.addInfo(NepoInfo.error("ERROR_MISSING_PARAMETER"));
    this.errorCount++;
} else {
    repeatStmt.getExpr().visit(super);
}
```
where super is a super visitor implementing all the emptiness checks. It could be implemented here, (this), but it would be better to factor this out in a super class and have only concrete implementations for each class.

TLDR:
- AbstractSimValidatorVisitor.visitRepeatStmt() should either not call super or not re-implement the super method.
- AbstractProgramValidatorVisitor.visitRepeatStmt() should not implement the checking of sub-expressions, instead in should check for emptiness of its input expression and in case it is present let it be visited. Visitors implementing the check of the sub-expression should check for the emptiness of its inputs and so on.

**Comment by rbudde** [comment]
should be solved when we do the complete rewrite of the empty-expr-logic and the corresponding visitors. Probably it can be integrated in the typecheck visitor, which then should include the empty-expr-detection, too.
